.github/workflows/cmake.yml

```diff
@@ -49,14 +60,30 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v3
 
+    - name: Install packages (Windows)
+      if: runner.os == 'Windows'
+      run: |
+        choco install --no-progress ninja ${{ matrix.packages }}
+
     - name: Generate project files
-      run: cmake -S ${{ matrix.src-dir || '../zlib' }} -B ${{ matrix.build-dir || '../build' }} ${{ matrix.cmake-args }} -D CMAKE_BUILD_TYPE=${{ matrix.build-config || 'Release' }}
+      run: cmake -S ${{ matrix.src-dir || '.' }} -B ${{ matrix.build-dir || '.' }} ${{ matrix.cmake-args }} -D CMAKE_BUILD_TYPE=${{ matrix.build-config || 'Release' }}
       env:
         CC: ${{ matrix.compiler }}
+        CFLAGS: ${{ matrix.cflags }}
 
     - name: Compile source code
-      run: cmake --build ${{ matrix.build-dir || '../build' }} --config ${{ matrix.build-config || 'Release' }}
+      run: cmake --build ${{ matrix.build-dir || '.' }} --config ${{ matrix.build-config || 'Release' }}
 
     - name: Run test cases
       run: ctest -C Release --output-on-failure --max-width 120
-      working-directory: ${{ matrix.build-dir || '../build' }}
+      working-directory: ${{ matrix.build-dir || '.' }}
+
+    - name: Upload build errors
+      uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: ${{ matrix.name }} (cmake)
+        path: |
+          **/CMakeFiles/CMakeOutput.log
+          **/CMakeFiles/CMakeError.log
+        retention-days: 30
```

Comment:
I don't think this is needed.
The results of a run are already stored for a while. And it doesn't seem like any other workflow is downloading the artifact, right?

Reply:
(If we do end up using this, v3 is out. No reason to stick with v2.)

Reply:
I have changed it to v3. It is needed because errors on compiler checks that are part of configure or CMake are not reporting through the build log that is displayed on GHA. These artifacts are only uploaded when there is a workflow failure and then are only retained for a certain number of days - I've changed it to 7.